virtualsmartcard/src/vpicc/virtualsmartcard/SEutils.py

```diff
@@ -694,31 +693,31 @@ def generate_public_key_pair(self, p1, p2, data):
             raise SwError(SW["ERR_CONDITIONNOTSATISFIED"])
 
         if p1 & 0x01 == 0x00:  # Generate key
-            PublicKey = c_class.generate(self.dst.keylength, rnd.get_bytes)
+            PublicKey = c_class.generate(self.dst.keylength)
             self.dst.key = PublicKey
         else:
             pass  # Read key
 
         # Encode keys
         if cipher == "RSA":
             # Public key
-            n = inttostring(PublicKey.__getstate__()['n'])
-            e = inttostring(PublicKey.__getstate__()['e'])
+            n = inttostring(PublicKey.n)
+            e = inttostring(PublicKey.e)
             pk = ((0x81, len(n), n), (0x82, len(e), e))
             result = bertlv_pack(pk)
             # Private key
-            d = PublicKey.__getstate__()['d']
+            d = PublicKey.d
         elif cipher == "DSA":
             # DSAParams
-            p = inttostring(PublicKey.__getstate__()['p'])
-            q = inttostring(PublicKey.__getstate__()['q'])
-            g = inttostring(PublicKey.__getstate__()['g'])
+            p = inttostring(PublicKey.p)
```

Comment:
The changes from __getstate__ to direct access is that in cryptodome  the access by __getstate__ raises a pickling exception